lnwallet/chanfunding/coin_select.go

```diff
@@ -270,13 +280,29 @@ func CoinSelectUpToAmount(feeRate chainfee.SatPerKWeight, maxAmt,
 	selectedCoins, changeAmt, err := CoinSelect(
 		feeRate, maxAmt, dustLimit, coins,
 	)
+	if err == nil {
+		// Validate that the selected amount will leave the required reserved
+		// value in the balance.
+		selected := btcutil.Amount(0)
+		for _, coin := range selectedCoins {
+			selected += btcutil.Amount(coin.Value)
+		}
+
+		// We subtract the change amount to obtain the actual amount that is
+		// selected.
+		selected -= changeAmt
+
+		if balance-selected < reserved {
+			err = &errReservedValueInvalidated{reserved}
```

Comment:
So this error is only needed to signal the code below to use the `CoinSelectSubtractFees`?
I think this can be solved differently, without needing to introduce a new error. Will comment on next commit as there's some refactoring going on there.